doc/_extensions/table_from_rows.py

```diff
@@ -126,29 +110,114 @@ def run(self):
         if rows is not None:
             rows = [r.strip() for r in rows.split(',') if r.strip() != '']
         else:
-            rows = []
+            raise self.severe('No "rows" content provided for "%s" '
+                              % self.name)
+
+        # join all rows for uniqueness, using the order given in
+        # the :rows: option.
+        all_rows = {}
+        all_rows.update(dict.fromkeys(rows))
+        raw = '\n'.join(self._load_header_and_rows(path, header,
+                                                   all_rows.keys()))
+        lines = statemachine.string2lines(raw)
+        self.state_machine.insert_input(lines, path)
+        return []
 
-        if 'sample-yaml-rows' in self.options:
-            source = self.state_machine.input_lines.source(
-                 self.lineno - self.state_machine.input_offset - 1)
-            source_dir = os.path.dirname(os.path.abspath(source))
-            sample_yaml_rows = self._rows_from_sample_yaml(source_dir)
+class TableFromSampleYaml(TableFromRows):
+
+    option_spec = {'header': directives.unchanged}
+
+    @staticmethod
+    def _merge_boards(boards):
+        """Converts pairs of similar board types to one table entry."""
+
+        merge_pairs = [
+            ("nrf9160dk_nrf9160", "nrf9160dk_nrf9160_ns", "nrf9160dk_nrf9160_and_ns"),
+            ("nrf5340dk_nrf5340_cpuapp", "nrf5340dk_nrf5340_cpuapp_ns", "nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns"),
+            ("nrf5340dk_nrf5340_cpuapp", "nrf5340dk_nrf5340_cpunet", "nrf5340dk_nrf5340_cpuapp_and_cpunet"),
+            ("thingy53_nrf5340_cpuapp", "thingy53_nrf5340_cpuapp_ns", "thingy53_nrf5340_cpuapp_and_cpuapp_ns"),
+        ]
+
+        for b1, b2, merged in merge_pairs:
+            if b1 in boards and b2 in boards:
+                boards.remove(b1)
+                boards.remove(b2)
+                boards.add(merged)
+
+    def _rows_from_sample_yaml(self, path):
+        """Search for a sample.yaml file and return a union of all relevant boards.
+
+        Boards are retrieved from the integration_platforms sections in the file.
+        If the file is not found and the document folder is named "doc", the
+        parent folder is searched. Should a sample.yaml still not be found, all
+        subfolders are searched and every found sample.yaml is used.
+
+        nrf51 and qemu boards are ignored."""
+
+        rel_path = os.path.relpath(path, self.env.srcdir).lstrip('..' + os.path.sep)
+        dir_path = Path(self.config.table_from_rows_base_dir) / rel_path
```

Comment:
this is hard to read/follow, maybe (not tesed_:

```python
rel_path = os.path.relpath(path, self.env.srcdir)
dir_path = Path(self.config.table_from_rows_base_dir).parent / rel_path
```

Reply:
This piece of code is taking care of an edge case where the supplied path did not point to the build directory. It has been restructured and added comments to make it easier to follow.